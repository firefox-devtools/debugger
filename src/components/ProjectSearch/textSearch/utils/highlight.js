// Maybe reuse file search's functions?
import React from "react";

export function highlightMatches(lineMatch) {
  const { value, column, match } = lineMatch;
  const len = match.length;

  return (
    <span className="line-value">
      <span className="line-match" key={0}>
        {value.slice(0, column)}
      </span>
      <span className="query-match" key={1}>
        {value.substr(column, len)}
      </span>
      <span className="line-match" key={2}>
        {value.slice(column + len, value.length)}
      </span>
    </span>
  );
}
