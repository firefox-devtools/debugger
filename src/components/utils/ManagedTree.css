.tree {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -o-user-select: none;
  user-select: none;

  flex: 1;
  white-space: nowrap;
  overflow: auto;
}

.tree button {
  display: block;
}

.tree .node {
  padding: 2px 5px;
  position: relative;
}

.tree .node.focused {
  color: white;
  background-color: var(--theme-selection-background);
}

html:not([dir="rtl"]) .tree .node > div {
  margin-left: 10px;
}

html[dir="rtl"] .tree .node > div {
  margin-right: 10px;
}

.tree .node.focused svg {
  fill: white;
}

.tree-node button {
  position: fixed;
}
