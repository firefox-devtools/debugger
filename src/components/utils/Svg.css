.worker,
.folder {
    position: relative;
    top: 2px;
}

.arrow,
.folder,
.domain,
.file,
.worker,
.refresh,
.add-button {
    fill: var(--theme-splitter-color);
}

.domain,
.file,
.worker,
.refresh,
.add-button {
    position: relative;
    top: 1px;
}

.domain svg,
.folder svg,
.worker svg,
.refresh svg,
.add-button svg {
    width: 15px;
    fill: currentColor;
}

.file svg {
    width: 13px;
}

html:not([dir="rtl"]) .file svg,
html:not([dir="rtl"]) .domain svg,
html:not([dir="rtl"]) .folder svg,
html:not([dir="rtl"]) .refresh svg,
html:not([dir="rtl"]) .worker svg {
    margin-right: 5px;
}

html[dir="rtl"] .file svg,
html[dir="rtl"] .domain svg,
html[dir="rtl"] .folder svg,
html[dir="rtl"] .worker svg,
html[dir="rtl"] .refresh svg {
    margin-left: 5px;
}
