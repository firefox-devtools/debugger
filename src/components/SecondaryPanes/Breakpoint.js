/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

// @flow
import React, { Component } from "react";
import ReactDOM from "react-dom";
import classnames from "classnames";

import CloseButton from "../shared/Button/Close";

import { createEditor } from "../../utils/breakpoint";
import { features } from "../../utils/prefs";

import type { LocalBreakpoint } from "./Breakpoints";
import type SourceEditor from "../../utils/editor/source-editor";

type Props = {
  breakpoint: LocalBreakpoint,
  onClick: Function,
  onContextMenu: Function,
  onChange: Function,
  onCloseClick: Function
};

function getBreakpointLocation(source, line, column) {
  const isWasm = source && source.isWasm;
  const columnVal = features.columnBreakpoints && column ? `:${column}` : "";
  const bpLocation = isWasm
    ? `0x${line.toString(16).toUpperCase()}`
    : `${line}${columnVal}`;

  return bpLocation;
}

class Breakpoint extends Component<Props> {
  editor: SourceEditor;

  componentDidMount() {
    this.setupEditor();
  }

  componentDidUpdate() {
    this.setupEditor();
  }

  componentWillUnmount() {
    if (this.editor) {
      this.editor.destroy();
      this.editor = null;
    }
  }

  shouldComponentUpdate(nextProps: Props) {
    const prevBreakpoint = this.props.breakpoint;
    const nextBreakpoint = nextProps.breakpoint;

    return (
      !prevBreakpoint ||
      (prevBreakpoint.text != nextBreakpoint.text ||
        prevBreakpoint.disabled != nextBreakpoint.disabled ||
        prevBreakpoint.condition != nextBreakpoint.condition ||
        prevBreakpoint.hidden != nextBreakpoint.hidden ||
        prevBreakpoint.isCurrentlyPaused != nextBreakpoint.isCurrentlyPaused)
    );
  }

  getBreakpointText() {
    const { breakpoint } = this.props;

    return breakpoint.condition || breakpoint.text;
  }

  setupEditor() {
    if (this.editor) {
      return;
    }

    this.editor = createEditor(this.getBreakpointText());

    // disables the default search shortcuts
    // $FlowIgnore
    this.editor._initShortcuts = () => {};

    const node = ReactDOM.findDOMNode(this);
    if (node instanceof HTMLElement) {
      const mountNode = node.querySelector(".breakpoint-label");
      if (node instanceof HTMLElement) {
        // $FlowIgnore
        mountNode.innerHTML = "";
        this.editor.appendToLocalElement(mountNode);
        this.editor.codeMirror.on("mousedown", (_, e) => e.preventDefault());
      }
    }
  }

  renderCheckbox() {
    const { onChange, breakpoint } = this.props;
    const { disabled } = breakpoint;

    return (
      <input
        type="checkbox"
        className="breakpoint-checkbox"
        checked={!disabled}
        onChange={onChange}
        onClick={ev => ev.stopPropagation()}
      />
    );
  }

  renderText() {
    const text = this.getBreakpointText();

    return (
      <label className="breakpoint-label" title={text}>
        {text}
      </label>
    );
  }

  renderLineClose() {
    const { breakpoint, onCloseClick } = this.props;
    const { line, column } = breakpoint.location;

    return (
      <div className="breakpoint-line-close">
        <div className="breakpoint-line">
          {getBreakpointLocation(breakpoint.source, line, column)}
        </div>
        <CloseButton
          handleClick={onCloseClick}
          tooltip={L10N.getStr("breakpoints.removeBreakpointTooltip")}
        />
      </div>
    );
  }

  render() {
    const { breakpoint, onClick, onContextMenu } = this.props;

    const locationId = breakpoint.locationId;
    const isCurrentlyPaused = breakpoint.isCurrentlyPaused;
    const isDisabled = breakpoint.disabled;
    const isConditional = !!breakpoint.condition;

    return (
      <div
        className={classnames({
          breakpoint,
          paused: isCurrentlyPaused,
          disabled: isDisabled,
          "is-conditional": isConditional
        })}
        key={locationId}
        onClick={onClick}
        onContextMenu={onContextMenu}
      >
        {this.renderCheckbox()}
        {this.renderText()}
        {this.renderLineClose()}
      </div>
    );
  }
}

export default Breakpoint;
