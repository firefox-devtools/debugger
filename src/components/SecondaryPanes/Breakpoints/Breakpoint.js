/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

// @flow

import React, { PureComponent } from "react";
import { connect } from "../../../utils/connect";
import { createSelector } from "reselect";
import classnames from "classnames";
import actions from "../../../actions";
import { memoize } from "lodash";

import showContextMenu from "./BreakpointsContextMenu";
import { CloseButton } from "../../shared/Button";

import {
  getLocationWithoutColumn,
  getSelectedText,
  makeBreakpointId
} from "../../../utils/breakpoint";
import { getSelectedLocation } from "../../../utils/source-maps";
import { features } from "../../../utils/prefs";
import { getEditor } from "../../../utils/editor";

import type {
  Breakpoint as BreakpointType,
  Frame,
  Source,
  SourceLocation
} from "../../../types";

type FormattedFrame = Frame & {
  selectedLocation: SourceLocation
};

import {
  getBreakpointsList,
  getSelectedFrame,
  getSelectedSource
} from "../../../selectors";

type Props = {
  breakpoint: BreakpointType,
  breakpoints: BreakpointType[],
  selectedSource: Source,
  source: Source,
  frame: FormattedFrame,
  enableBreakpoint: typeof actions.enableBreakpoint,
  removeBreakpoint: typeof actions.removeBreakpoint,
  removeBreakpoints: typeof actions.removeBreakpoints,
  removeAllBreakpoints: typeof actions.removeAllBreakpoints,
  disableBreakpoint: typeof actions.disableBreakpoint,
  setBreakpointOptions: typeof actions.setBreakpointOptions,
  toggleAllBreakpoints: typeof actions.toggleAllBreakpoints,
  toggleBreakpoints: typeof actions.toggleBreakpoints,
  toggleDisabledBreakpoint: typeof actions.toggleDisabledBreakpoint,
  openConditionalPanel: typeof actions.openConditionalPanel,
  selectSpecificLocation: typeof actions.selectSpecificLocation
};

class Breakpoint extends PureComponent<Props> {
  onContextMenu = e => {
    showContextMenu({ ...this.props, contextMenuEvent: e });
  };

  get selectedLocation() {
    const { breakpoint, selectedSource } = this.props;
    return getSelectedLocation(breakpoint, selectedSource);
  }

  onDoubleClick = () => {
    const { breakpoint, openConditionalPanel } = this.props;
    if (breakpoint.options.condition) {
      openConditionalPanel(this.selectedLocation);
    }
  };

  selectBreakpoint = event => {
    event.preventDefault();
    const { selectSpecificLocation } = this.props;
    selectSpecificLocation(this.selectedLocation);
  };

  removeBreakpoint = event => {
    const { removeBreakpoint, breakpoint } = this.props;
    event.stopPropagation();
    removeBreakpoint(breakpoint);
  };

  handleBreakpointCheckbox = () => {
    const { breakpoint, enableBreakpoint, disableBreakpoint } = this.props;
    if (breakpoint.disabled) {
      enableBreakpoint(breakpoint);
    } else {
      disableBreakpoint(breakpoint);
    }
  };

  isCurrentlyPausedAtBreakpoint() {
    const { frame } = this.props;
    if (!frame) {
      return false;
    }

    const bpId = features.columnBreakpoints
      ? makeBreakpointId(this.selectedLocation)
      : getLocationWithoutColumn(this.selectedLocation);
    const frameId = features.columnBreakpoints
      ? makeBreakpointId(frame.selectedLocation)
      : getLocationWithoutColumn(frame.selectedLocation);
    return bpId == frameId;
  }

  getBreakpointLocation() {
    const { source } = this.props;
    const { column, line } = this.selectedLocation;

    const isWasm = source && source.isWasm;
    const columnVal = features.columnBreakpoints && column ? `:${column}` : "";
    const bpLocation = isWasm
      ? `0x${line.toString(16).toUpperCase()}`
      : `${line}${columnVal}`;

    return bpLocation;
  }

  getBreakpointText() {
    const { breakpoint, selectedSource } = this.props;
    const { condition, logValue } = breakpoint.options;
    return logValue || condition || getSelectedText(breakpoint, selectedSource);
  }

  highlightText = memoize(
    (text = "", editor) => {
      if (!editor.CodeMirror) {
        return { __html: text };
      }

      const node = document.createElement("div");
      editor.CodeMirror.runMode(text, "application/javascript", node);
      return { __html: node.innerHTML };
    },
    (text, editor) => `${text} - ${editor.CodeMirror ? "editor" : ""}`
  );

  /* eslint-disable react/no-danger */
  render() {
    const { breakpoint } = this.props;
    const text = this.getBreakpointText();
    const editor = getEditor();
    return (
      <div
        className={classnames({
          breakpoint,
          paused: this.isCurrentlyPausedAtBreakpoint(),
          disabled: breakpoint.disabled,
          "is-conditional": !!breakpoint.options.condition,
          "is-log": !!breakpoint.options.logValue
        })}
        onClick={this.selectBreakpoint}
        onDoubleClick={this.onDoubleClick}
        onContextMenu={this.onContextMenu}
      >
        <input
          id={breakpoint.id}
          type="checkbox"
          className="breakpoint-checkbox"
          checked={!breakpoint.disabled}
          onChange={this.handleBreakpointCheckbox}
          onClick={ev => ev.stopPropagation()}
        />
        <label
          htmlFor={breakpoint.id}
          className="breakpoint-label cm-s-mozilla devtools-monospace"
          onClick={this.selectBreakpoint}
          title={text}
        >
          <span dangerouslySetInnerHTML={this.highlightText(text, editor)} />
        </label>
        <div className="breakpoint-line-close">
          <div className="breakpoint-line devtools-monospace">
            {this.getBreakpointLocation()}
          </div>
          <CloseButton
            handleClick={e => this.removeBreakpoint(e)}
            tooltip={L10N.getStr("breakpoints.removeBreakpointTooltip")}
          />
        </div>
      </div>
    );
  }
}

const getFormattedFrame = createSelector(
  getSelectedSource,
  getSelectedFrame,
  (selectedSource: ?Source, frame: ?Frame): ?FormattedFrame => {
    if (!frame) {
      return null;
    }

    return {
      ...frame,
      selectedLocation: getSelectedLocation(frame, selectedSource)
    };
  }
);

const mapStateToProps = state => ({
  breakpoints: getBreakpointsList(state),
  frame: getFormattedFrame(state)
});

export default connect(
  mapStateToProps,
  {
    enableBreakpoint: actions.enableBreakpoint,
    removeBreakpoint: actions.removeBreakpoint,
    removeBreakpoints: actions.removeBreakpoints,
    removeAllBreakpoints: actions.removeAllBreakpoints,
    disableBreakpoint: actions.disableBreakpoint,
    selectSpecificLocation: actions.selectSpecificLocation,
    setBreakpointOptions: actions.setBreakpointOptions,
    toggleAllBreakpoints: actions.toggleAllBreakpoints,
    toggleBreakpoints: actions.toggleBreakpoints,
    toggleDisabledBreakpoint: actions.toggleDisabledBreakpoint,
    openConditionalPanel: actions.openConditionalPanel
  }
)(Breakpoint);
