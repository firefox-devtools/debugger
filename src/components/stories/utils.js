const renderComponentFromFixture =
      require("../../test/utils/renderComponentFromFixture");

const { storiesOf } = require("@kadira/storybook");

module.exports = {
  renderComponent: renderComponentFromFixture,
  storiesOf
};
