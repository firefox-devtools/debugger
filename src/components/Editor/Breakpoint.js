// @flow
import React, { Component } from "react";
import { isEnabled } from "devtools-config";
import ReactDOM from "react-dom";

import classnames from "classnames";
import Svg from "../shared/Svg";

import { getDocument, showSourceText, toEditorLine } from "../../utils/editor";

const breakpointSvg = document.createElement("div");
ReactDOM.render(<Svg name="breakpoint" />, breakpointSvg);

function makeMarker(isDisabled: boolean) {
  const bp = breakpointSvg.cloneNode(true);
  bp.className = classnames("editor new-breakpoint", {
    "breakpoint-disabled": isDisabled,
    "folding-enabled": isEnabled("codeFolding")
  });

  return bp;
}

class Breakpoint extends Component {
  props: {
    breakpoint: Object,
    selectedSource: Object,
    editor: Object
  };

  addBreakpoint: Function;

  constructor() {
    super();
    this.addBreakpoint = this.addBreakpoint.bind(this);
  }

  addBreakpoint() {
    const { breakpoint, editor, selectedSource } = this.props;

    // Hidden Breakpoints are never rendered on the client
    if (breakpoint.hidden) {
      return;
    }

    // NOTE: we need to wait for the breakpoint to be loaded
    // to get the generated location
    if (!selectedSource || breakpoint.loading) {
      return;
    }

    const sourceId = selectedSource.get("id");
    const line = toEditorLine(sourceId, breakpoint.location.line);

    showSourceText(editor, selectedSource.toJS());

    editor.codeMirror.setGutterMarker(
      line,
      "breakpoints",
      makeMarker(breakpoint.disabled)
    );

    editor.codeMirror.addLineClass(line, "line", "new-breakpoint");
    if (breakpoint.condition) {
      editor.codeMirror.addLineClass(line, "line", "has-condition");
    } else {
      editor.codeMirror.removeLineClass(line, "line", "has-condition");
    }
  }

  shouldComponentUpdate(nextProps: any) {
    const { editor, breakpoint, selectedSource } = this.props;
    return (
      editor !== nextProps.editor ||
      breakpoint.disabled !== nextProps.breakpoint.disabled ||
      breakpoint.hidden !== nextProps.breakpoint.hidden ||
      breakpoint.condition !== nextProps.breakpoint.condition ||
      breakpoint.loading !== nextProps.breakpoint.loading ||
      selectedSource !== nextProps.selectedSource
    );
  }

  componentDidMount() {
    this.addBreakpoint();
  }

  componentDidUpdate() {
    this.addBreakpoint();
  }

  componentWillUnmount() {
    const { editor, breakpoint, selectedSource } = this.props;

    if (!selectedSource) {
      return;
    }

    if (breakpoint.loading) {
      return;
    }

    const sourceId = selectedSource.get("id");
    const doc = getDocument(sourceId);
    if (!doc) {
      return;
    }

    const line = toEditorLine(sourceId, breakpoint.location.line);

    // NOTE: when we upgrade codemirror we can use `doc.setGutterMarker`
    if (doc.setGutterMarker) {
      doc.setGutterMarker(line, "breakpoints", null);
    } else {
      editor.codeMirror.setGutterMarker(line, "breakpoints", null);
    }

    doc.removeLineClass(line, "line", "new-breakpoint");
    doc.removeLineClass(line, "line", "has-condition");
  }

  render() {
    return null;
  }
}

export default Breakpoint;
