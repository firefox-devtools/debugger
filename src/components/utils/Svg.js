/**
 * This file maps the SVG React Components in the assets/images directory.
 */
const Svg = require("../../../assets/images/Svg");

require("./Svg.css");

module.exports = Svg;
