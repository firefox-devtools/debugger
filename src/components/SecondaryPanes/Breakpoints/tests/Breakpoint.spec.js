/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

import React from "react";
import { shallow } from "enzyme";

import Breakpoint from "../Breakpoint";
import { makeSource, makeOriginalSource } from "../../../../utils/test-head";

describe("Breakpoint", () => {
  it("simple", () => {
    const { component } = render();
    expect(component).toMatchSnapshot();
  });

  it("disabled", () => {
    const { component } = render({
      breakpoint: makeBreakpoint({ disabled: true })
    });
    expect(component).toMatchSnapshot();
  });

  it("selected source is original", () => {
    const { component } = render({
      source: makeOriginalSource("foo"),
      selectedSource: makeOriginalSource("foo")
    });
    expect(component).toMatchSnapshot();
  });

  it("paused at a generatedLocation", () => {
    const { component } = render({
      frame: { generatedLocation, location }
    });
    expect(component).toMatchSnapshot();
  });

  it("paused at an original location", () => {
    const { component } = render({
      frame: { location, generatedLocation },
      selectedSource: makeOriginalSource("foo")
    });
    expect(component).toMatchSnapshot();
  });

  it("paused at a different", () => {
    const { component } = render({
      frame: { location, generatedLocation: { ...generatedLocation, line: 14 } }
    });
    expect(component).toMatchSnapshot();
  });
});

const generatedLocation = { sourceId: "foo", line: 53, column: 73 };
const location = { sourceId: "foo/original", line: 5, column: 7 };
function render(overrides = {}) {
  const props = generateDefaults(overrides);
  const component = shallow(<Breakpoint.WrappedComponent {...props} />);
  const defaultState = component.state();
  const instance = component.instance();

  return { component, props, defaultState, instance };
}

function makeBreakpoint(overrides = {}) {
  return {
    generatedLocation,
    location,
    disabled: false,
    ...overrides
  };
}

function generateDefaults(overrides) {
  const source = makeSource("foo");
  const breakpoint = makeBreakpoint();
  const selectedSource = makeSource("foo");
  return {
    source,
    breakpoint,
    selectedSource,
    frame: null,
    ...overrides
  };
}
