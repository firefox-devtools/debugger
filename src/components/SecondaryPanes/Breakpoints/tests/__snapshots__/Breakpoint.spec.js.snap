// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breakpoint disabled 1`] = `
<div
  className="breakpoint disabled"
  onClick={[Function]}
  onContextMenu={[Function]}
  onDoubleClick={[Function]}
>
  <input
    checked={false}
    className="breakpoint-checkbox"
    onChange={[Function]}
    onClick={[Function]}
    type="checkbox"
  />
  <label
    className="breakpoint-label cm-s-mozilla devtools-monospace"
    onClick={[Function]}
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
  </label>
  <div
    className="breakpoint-line-close"
  >
    <div
      className="breakpoint-line devtools-monospace"
    >
      53
    </div>
    <CloseButton
      handleClick={[Function]}
      tooltip="Remove breakpoint"
    />
  </div>
</div>
`;

exports[`Breakpoint paused at a different 1`] = `
<div
  className="breakpoint"
  onClick={[Function]}
  onContextMenu={[Function]}
  onDoubleClick={[Function]}
>
  <input
    checked={true}
    className="breakpoint-checkbox"
    onChange={[Function]}
    onClick={[Function]}
    type="checkbox"
  />
  <label
    className="breakpoint-label cm-s-mozilla devtools-monospace"
    onClick={[Function]}
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
  </label>
  <div
    className="breakpoint-line-close"
  >
    <div
      className="breakpoint-line devtools-monospace"
    >
      53
    </div>
    <CloseButton
      handleClick={[Function]}
      tooltip="Remove breakpoint"
    />
  </div>
</div>
`;

exports[`Breakpoint paused at a generatedLocation 1`] = `
<div
  className="breakpoint paused"
  onClick={[Function]}
  onContextMenu={[Function]}
  onDoubleClick={[Function]}
>
  <input
    checked={true}
    className="breakpoint-checkbox"
    onChange={[Function]}
    onClick={[Function]}
    type="checkbox"
  />
  <label
    className="breakpoint-label cm-s-mozilla devtools-monospace"
    onClick={[Function]}
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
  </label>
  <div
    className="breakpoint-line-close"
  >
    <div
      className="breakpoint-line devtools-monospace"
    >
      53
    </div>
    <CloseButton
      handleClick={[Function]}
      tooltip="Remove breakpoint"
    />
  </div>
</div>
`;

exports[`Breakpoint paused at an original location 1`] = `
<div
  className="breakpoint paused"
  onClick={[Function]}
  onContextMenu={[Function]}
  onDoubleClick={[Function]}
>
  <input
    checked={true}
    className="breakpoint-checkbox"
    onChange={[Function]}
    onClick={[Function]}
    type="checkbox"
  />
  <label
    className="breakpoint-label cm-s-mozilla devtools-monospace"
    onClick={[Function]}
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
  </label>
  <div
    className="breakpoint-line-close"
  >
    <div
      className="breakpoint-line devtools-monospace"
    >
      5
    </div>
    <CloseButton
      handleClick={[Function]}
      tooltip="Remove breakpoint"
    />
  </div>
</div>
`;

exports[`Breakpoint simple 1`] = `
<div
  className="breakpoint"
  onClick={[Function]}
  onContextMenu={[Function]}
  onDoubleClick={[Function]}
>
  <input
    checked={true}
    className="breakpoint-checkbox"
    onChange={[Function]}
    onClick={[Function]}
    type="checkbox"
  />
  <label
    className="breakpoint-label cm-s-mozilla devtools-monospace"
    onClick={[Function]}
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
  </label>
  <div
    className="breakpoint-line-close"
  >
    <div
      className="breakpoint-line devtools-monospace"
    >
      53
    </div>
    <CloseButton
      handleClick={[Function]}
      tooltip="Remove breakpoint"
    />
  </div>
</div>
`;
